pkg/tenant/auth.go

```diff
@@ -139,13 +142,53 @@ func (s *Server) Login(c *gin.Context) {
 		Password: params.Password,
 	}
 
+	ok := hasInviteToken(params.InviteToken)
+
+	if ok {
+		req.InviteToken = params.InviteToken
+	}
+
 	var reply *qd.LoginReply
 	if reply, err = s.quarterdeck.Login(c.Request.Context(), req); err != nil {
 		sentry.Debug(c).Err(err).Msg("tracing quarterdeck error in tenant")
 		api.ReplyQuarterdeckError(c, err)
 		return
 	}
 
+	// Update the status of a member with an invite token to Confirmed.
+	if ok && reply.AccessToken != "" {
+		// Parse access token to get the orgID.
+		var claims *jwt.RegisteredClaims
+		if claims, err = tokens.ParseUnverified(reply.AccessToken); err != nil {
+			sentry.Error(c).Err(err).Msg("could not parse access token from the claims")
+			c.JSON(http.StatusInternalServerError, api.ErrorResponse("user claims unavailable"))
+			return
+		}
+
+		var orgID ulid.ULID
+		if orgID, err = ulid.Parse(claims.ID); err != nil {
+			sentry.Error(c).Err(err).Msg("could not parse orgID from access token")
+			c.JSON(http.StatusInternalServerError, api.ErrorResponse("member not found"))
+			return
+		}
+
+		// Get member from the database by their email.
+		var member *db.Member
+		if member, err = db.GetMemberByEmail(c, orgID, params.Email); err != nil {
+			sentry.Error(c).Err(err).Msg("could not get member from the database")
+			c.JSON(http.StatusInternalServerError, api.ErrorResponse("member not found"))
+			return
+		}
+
+		// Update member status to Confirmed.
+		member.Status = db.MemberStatusConfirmed
+		if err = db.UpdateMember(c, member); err != nil {
+			sentry.Error(c).Err(err).Msg("could not update member in the database")
+			c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not update member"))
+			return
+		}
```

Comment:
I think we also need to include the same delete/create logic you had in register since the same situation applies: if the user creates a new account after the invite was already issued then the member ID needs to be updated, which in trtl means creating a new record.